### PR TITLE
Add detail configaudit info as prometheus metric

### LINKED ISSUE #1148: Expose detail info for config audit report as  Prometheus metrics
is it possible to get configAudit ID (KSV) in Prometheus, would like to have them in grafana, thank you

### PR DESCRIPTION
**Pull request by geoffrey1330**
## Related issues
- Close #1148 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).

### CODE REVIEW

File: pkg/metrics/collector.go [hunk 1]
```diff
@@ -47,6 +47,11 @@ const (
 	secret_rule_id  = "secret_rule_id"
 	secret_target   = "secret_target"
 	secret_title    = "secret_title"
+	//config audit
+	config_audit_id          = "config_audit_id"
+	config_audit_title       = "config_audit_title"
+	config_audit_description = "config_audit_description"
+	config_audit_category    = "config_audit_category"
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:54]: probably it will be good to add `success` field bool

File: pkg/metrics/collector.go [hunk 2]
```diff
@@ -591,6 +621,39 @@ func (c *ResourcesMetricsCollector) collectConfigAuditReports(ctx context.Contex
...
+			if c.Config.MetricsConfigAuditInfo {
+				labelValues[0] = r.Namespace
+				labelValues[1] = r.Name
+				labelValues[2] = r.Labels[trivyoperator.LabelResourceKind]
+				labelValues[3] = r.Labels[trivyoperator.LabelResourceName]
+				for _, config := range r.Report.Checks {
+					labelValues[4] = config.ID
+					labelValues[5] = config.Title
+					labelValues[6] = config.Description
+					labelValues[7] = config.Category
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:642]: Add success field

File: pkg/metrics/collector_test.go [hunk 3]
```diff
@@ -472,16 +484,16 @@ var _ = Describe("ResourcesMetricsCollector", func() {
 			const expected = `
         # HELP trivy_resource_configaudits Number of failing resource configuration auditing checks
         # TYPE trivy_resource_configaudits gauge
-        trivy_resource_configaudits{name="configmap-test",namespace="some-ns",resource_kind="ConfigMap",resource_name="test",severity="Critical"} 0
```
Comment (chen-keinan) [inline pkg/metrics/collector_test.go:475]: `summary report` test should remain , info test should be separate test

File: pkg/metrics/collector_test.go [hunk 4]
```diff
@@ -450,20 +462,20 @@ var _ = Describe("ResourcesMetricsCollector", func() {
...
```
Comment (chen-keinan) [inline pkg/metrics/collector_test.go:453]: `summary report` test should remain , info test should be separate test

File: pkg/metrics/collector_test.go (hunk 3 above)
Comment (geoffrey1330) [reply pkg/metrics/collector_test.go:475]: @chen-keinan I'm not to clear on what changes to make here

File: pkg/metrics/collector_test.go (hunk 4 above)
Comment (geoffrey1330) [reply pkg/metrics/collector_test.go:453]: @chen-keinan I'm not to clear on what changes to make here

File: pkg/metrics/collector.go [hunk 5]
```diff
@@ -52,6 +52,7 @@ const (
 	config_audit_title       = "config_audit_title"
 	config_audit_description = "config_audit_description"
 	config_audit_category    = "config_audit_category"
+	config_audit_success     = false
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:55]: config_audit_success field name should be equal to `config_audit_success`

Comment (geoffrey1330) [reply pkg/metrics/collector.go]: I've implemented this

File: pkg/metrics/collector_test.go (hunk 3 above)
Comment (chen-keinan) [reply pkg/metrics/collector_test.go:475]: I saw that override perv. summary tests  with new info tests.
just keep the old summary tests and add new tests for info data

Comment (geoffrey1330) [reply pkg/metrics/collector_test.go]: I've made the fix

Comment (geoffrey1330) [reply pkg/metrics/collector_test.go]: Resolved failing test

**Comment by chen-keinan** [comment]
@geoffrey1330 looks good, the issue is not just on `configaudit report` its also require to add metric info support for other reports.

are you planning to add it in a separate PR ?

**Comment by geoffrey1330** [comment]
> @geoffrey1330 looks good, the issue is not just on `configaudit report` its also require to add metric info support for other reports.
> 
> Are you planning to add it in a separate PR ?

Yes that will be much better If we create a new issue for the rest of the report

**Comment by chen-keinan** [comment]
> > @geoffrey1330 looks good, the issue is not just on `configaudit report` its also require to add metric info support for other reports.
> > Are you planning to add it in a separate PR ?
> 
> Yes that will be much better If we create a new issue for the rest of the report

OK then create new issues for others report type and associate it to the issue above

**Comment by geoffrey1330** [comment]
> > > @geoffrey1330 looks good, the issue is not just on `configaudit report` its also require to add metric info support for other reports.
> > > Are you planning to add it in a separate PR ?
> > 
> > 
> > Yes that will be much better If we create a new issue for the rest of the report
> 
> OK then create new issues for others report type and associate it to the issue above

Alright Will this PR be merged now?

**Comment by chen-keinan** [comment]
@geoffrey1330  lgtm 🚀  thanks you for the contribution

**Review by chen-keinan — APPROVED** [review]

File: pkg/metrics/collector_test.go [hunk 6]
```diff
@@ -467,7 +482,28 @@ var _ = Describe("ResourcesMetricsCollector", func() {
 				To(Succeed())
 		})
 
-		It("should produce correct metrics from target namespaces", func() {
+		It("should produce correct metrics with cluster scope - Info", func() {
+			const expected = `
+        # HELP trivy_resource_configaudits Number of failing resource configuration auditing checks
+        # TYPE trivy_resource_configaudits gauge
+        trivy_resource_configaudits{name="configmap-test",namespace="some-ns",resource_kind="ConfigMap",resource_name="test",config_audit_id="car2 Id",config_audit_title="car2 config audit title", config_audit_description="car2 description for config audit", config_audit_category="car2 category for config audit",config_audit_success="true",severity="Critical"} 0
```
Comment (chen-keinan) [inline pkg/metrics/collector_test.go:489]: it doesn't make sense metric should name  `trivy_configaudits_info` but I see it has summary metric name `trivy_resource_configaudits`

**Comment by chen-keinan** [comment]
@geoffrey1330 I have made some more tests and found issue , take a look above

File: pkg/metrics/collector_test.go (hunk 6 above)
Comment (geoffrey1330) [reply pkg/metrics/collector_test.go:489]: I will take a look at it

Comment (geoffrey1330) [reply pkg/metrics/collector_test.go:489]: @chen-keinan what do you suggest?

File: pkg/metrics/collector.go [hunk 7]
```diff
@@ -591,6 +623,40 @@ func (c *ResourcesMetricsCollector) collectConfigAuditReports(ctx context.Contex
...
+					labelValues[7] = config.Category
+					labelValues[8] = strconv.FormatBool(config.Success)
+				}
+				for i, label := range c.GetReportResourceLabels() {
+					labelValues[i+5] = r.Labels[label]
+				}
+				for severity, countFn := range c.configAuditSeverities {
+					labelValues[9] = severity
+					count := countFn(r.Report.Summary)
+					metrics <- prometheus.MustNewConstMetric(c.configAuditInfoDesc, prometheus.GaugeValue, float64(count), labelValues...)
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:653]: metric count for `collectConfigAuditInfoReports` is always `float64(1)`  as there is 1 count only for each entry

File: pkg/metrics/collector.go [hunk 8]
```diff
@@ -328,6 +349,12 @@ func buildMetricDescriptors(config trivyoperator.ConfigData) metricDescriptors {
 		configAuditLabels,
 		nil,
 	)
+	configAuditInfoDesc := prometheus.NewDesc(
+		prometheus.BuildFQName("trivy", "configaudits", "info"),
+		"Number of failing resource configuration auditing checks",
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:354]: it preferred to add the `info` at the end of description and fix the test , to distinguish between `info` and `summary`

Comment (geoffrey1330) [reply pkg/metrics/collector.go:354]: Alright Let me add me it now

**Comment by chen-keinan** [comment]
lgtm 🚀

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**